File: SDKs/googleapis.com/appengine/v1/go/pkg/models/operations/appengineappsfirewallingressruleslist.go
package operations

import (
"openapi/pkg/models/shared")

type AppengineAppsFirewallIngressRulesListPathParams struct {
    AppsID string `pathParam:"style=simple,explode=false,name=appsId"`
    
}

type AppengineAppsFirewallIngressRulesListQueryParams struct {
    DollarXgafv *shared.XgafvEnum `queryParam:"style=form,explode=true,name=$.xgafv"`
    AccessToken *string `queryParam:"style=form,explode=true,name=access_token"`
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Callback *string `queryParam:"style=form,explode=true,name=callback"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    MatchingAddress *string `queryParam:"style=form,explode=true,name=matchingAddress"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PageSize *int64 `queryParam:"style=form,explode=true,name=pageSize"`
    PageToken *string `queryParam:"style=form,explode=true,name=pageToken"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    UploadType *string `queryParam:"style=form,explode=true,name=uploadType"`
    UploadProtocol *string `queryParam:"style=form,explode=true,name=upload_protocol"`
    
}

type AppengineAppsFirewallIngressRulesListSecurityOption1 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type AppengineAppsFirewallIngressRulesListSecurityOption2 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type AppengineAppsFirewallIngressRulesListSecurityOption3 struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type AppengineAppsFirewallIngressRulesListSecurity struct {
    Option1 *AppengineAppsFirewallIngressRulesListSecurityOption1 `security:"option"`
    Option2 *AppengineAppsFirewallIngressRulesListSecurityOption2 `security:"option"`
    Option3 *AppengineAppsFirewallIngressRulesListSecurityOption3 `security:"option"`
    
}

type AppengineAppsFirewallIngressRulesListRequest struct {
    PathParams AppengineAppsFirewallIngressRulesListPathParams 
    QueryParams AppengineAppsFirewallIngressRulesListQueryParams 
    Security AppengineAppsFirewallIngressRulesListSecurity 
    
}

type AppengineAppsFirewallIngressRulesListResponse struct {
    ContentType string 
    ListIngressRulesResponse *shared.ListIngressRulesResponse 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/appengine/v1/typescript/src/sdk/models/operations/appengineappsfirewallingressruleslist.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class AppengineAppsFirewallIngressRulesListPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=appsId" })
  appsId: string;
}


export class AppengineAppsFirewallIngressRulesListQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=$.xgafv" })
  dollarXgafv?: shared.XgafvEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=access_token" })
  accessToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @Metadata({ data: "queryParam, style=form;explode=true;name=callback" })
  callback?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=matchingAddress" })
  matchingAddress?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=pageSize" })
  pageSize?: number;

  @Metadata({ data: "queryParam, style=form;explode=true;name=pageToken" })
  pageToken?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @Metadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=uploadType" })
  uploadType?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=upload_protocol" })
  uploadProtocol?: string;
}


export class AppengineAppsFirewallIngressRulesListSecurityOption1 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class AppengineAppsFirewallIngressRulesListSecurityOption2 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class AppengineAppsFirewallIngressRulesListSecurityOption3 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class AppengineAppsFirewallIngressRulesListSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, option=true" })
  option1?: AppengineAppsFirewallIngressRulesListSecurityOption1;

  @Metadata({ data: "security, option=true" })
  option2?: AppengineAppsFirewallIngressRulesListSecurityOption2;

  @Metadata({ data: "security, option=true" })
  option3?: AppengineAppsFirewallIngressRulesListSecurityOption3;
}


export class AppengineAppsFirewallIngressRulesListRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: AppengineAppsFirewallIngressRulesListPathParams;

  @Metadata()
  queryParams: AppengineAppsFirewallIngressRulesListQueryParams;

  @Metadata()
  security: AppengineAppsFirewallIngressRulesListSecurity;
}


export class AppengineAppsFirewallIngressRulesListResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listIngressRulesResponse?: shared.ListIngressRulesResponse;

  @Metadata()
  statusCode: number;
}
